internal/character/yanqing/trace.go

```diff
@@ -0,0 +1,46 @@
+package yanqing
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	A6            = "yanqing-a6"
+	A2 key.Attack = "yanqing-a2"
+)
+
+func init() {
+	modifier.Register(A6, modifier.Config{
+		StatusType: model.StatusType_STATUS_BUFF,
+		Stacking:   modifier.Refresh,
+		Duration:   2,
+	})
+}
+
+func (c *char) A2Listener(e event.AttackEnd) {
```

Comment:
This should be implemented as a listener for a modifier in my opinion, not as a subscriber to the `AttackEnd` event. Reasoning:
Current implementation listens for `AttackEnd`  events from ALL entities in the sim, not just yanqing, as subscribers listen for events from all sources that are of a specific event type. 
To fix this, you can instead use a modifier with the appropriate listener attached, as they inherently only care about events produced by their owner. You could also add a check as to whether or not the source of the attack is Yanqing, but I believe that implementation would be much messier. Former also matches more closely with the dm.